server/core/src/rodsServer.cpp

```diff
@@ -305,6 +313,126 @@ namespace
+
+        using log = irods::experimental::log;
+
+        for (auto&& entry : fs::directory_iterator{stacktrace_directory}) {
+            // Expected filename format:
+            //
+            //     <epoch_seconds>.<remaining_milliseconds>.<agent_pid>
+            //
+            // 1. Extract the timestamp from the filename and convert it to ISO8601 format.
+            // 2. Extract the agent pid from the filename.
+            const auto p = entry.path().generic_string();
+
+            // TODO std::string::rfind can be replaced with std::string::ends_with in C++20.
+            if (p.rfind(irods::STACKTRACE_NOT_READY_FOR_LOGGING_SUFFIX) != std::string::npos) { 
+                log::server::trace("Skipping [{}] ...", p);
+                continue;
+            }
+
+            auto slash_pos = p.rfind("/");
+
+            if (slash_pos == std::string::npos) {
+                log::server::trace("Skipping [{}]. No forward slash separator found.", p);
+                continue;
+            }
+
+            ++slash_pos;
+            const auto first_dot_pos = p.find(".", slash_pos);
+
+            if (first_dot_pos == std::string::npos) {
+                log::server::trace("Skipping [{}]. No dot separator found.", p);
+                continue;
+            }
+
+            const auto last_dot_pos = p.rfind(".");
+
+            if (last_dot_pos == std::string::npos || last_dot_pos == first_dot_pos) {
+                log::server::trace("Skipping [{}]. No dot separator found.", p);
+                continue;
+            }
+
+            const auto epoch_seconds = p.substr(slash_pos, first_dot_pos - slash_pos);
+            const auto remaining_millis = p.substr(first_dot_pos + 1, last_dot_pos - (first_dot_pos + 1));
+            const auto pid = p.substr(last_dot_pos + 1);
+            log::server::trace("epoch seconds = [{}], remaining millis = [{}], agent pid = [{}]",
+                               epoch_seconds, remaining_millis, pid);
+
+            try {
+                // Convert the epoch value to ISO8601 format.
+                log::server::trace("Converting epoch seconds to UTC timestamp ...");
+                using boost::chrono::system_clock;
+                using boost::chrono::time_fmt;
+                const auto tp = system_clock::from_time_t(std::stoll(epoch_seconds));
+                std::ostringstream utc_ss;
+                utc_ss << time_fmt(boost::chrono::timezone::utc, "%FT%T") << tp;
+
+                // Read the contents of the file.
+                std::ifstream file{p};
+                const auto stacktrace = boost::stacktrace::stacktrace::from_dump(file);
+                file.close();
+
+                // 3. Write the contents of the stacktrace file to syslog.
+                log::server::trace("Sending stacktrace buffer to syslog ...");
+                irods::experimental::log::server::error({
```

Comment:
It just occurred to me that maybe this should go under `server::critical` instead of `server::error`. Thoughts?

Reply:
i'd be okay with that - we don't use it very many places (1... 2?)

Reply:
that's reasonable.

i was planning on removing the trace log messages, but maybe we should keep them.

Reply:
done